File: competitive_programming/leetcode/01_python/242_valid_anagram.py
"""LeetCode 242 - Valid Anagram."""
def valid_anagram(s, t):
    return sorted(s) == sorted(t)


if __name__ == "__main__":
    assert valid_anagram("anagram", "nagaram") is True
    assert valid_anagram("rat", "car") is False
    print("242 valid anagram ok")

File: competitive_programming/leetcode/03_cpp/242_valid_anagram.cpp
#include <algorithm>
#include <cassert>
#include <cmath>
#include <cstdint>
#include <iostream>
#include <set>
#include <string>
#include <vector>

bool validAnagram(std::string s, std::string t) {
    std::sort(s.begin(), s.end());
    std::sort(t.begin(), t.end());
    return s == t;
}

int main() {
    assert(validAnagram("anagram", "nagaram") && !validAnagram("rat", "car"));
    std::cout << "242 valid anagram ok\n";
    return 0;
}
